jax/numpy/lax_numpy.py

```diff
@@ -441,6 +440,14 @@ def op(*args):
 logical_xor = _logical_op(np.logical_xor, lax.bitwise_xor)
 
 
+@_wraps(np.right_shift)
+def right_shift(x1, x2):
+  args = _promote_args(np.right_shift.__name__, x1, x2)
+  lax_fn = lax.shift_right_logical if \
+    np.issubdtype(x1.dtype, np.unsignedinteger) else lax.shift_right_arithmetic
```

Comment:
Change `x1.dtype` to `args[0].dtype`

Reply:
@jakevdp Btw. shouldn't we ideally have tests that cover these cases?

Reply:
The tests do cover mixed types, I think: https://github.com/google/jax/blob/15a9a70bb7a818cd84fc6e0ac45f6cde16a68845/tests/lax_numpy_test.py#L480-L486